### PR TITLE
JPERF-791 Generate docker image at runtime

### PR DESCRIPTION
**Pull request by mgrzaslewicz**
To avoid relying on pushing image to registry that we don't have control of
It happened already registry changed its policy and blocked pushing images

### CODE REVIEW

File: src/main/resources/docker/Dockerfile.template
Comment (pczuj) [inline src/main/resources/docker/Dockerfile.template:1]: Does it mean we are kinda forking this Dockerfile? Why do we want to keep copy of it here? Is it suggested way written somewhere in the `rastasheep` repository?

File: src/main/kotlin/com/atlassian/performance/tools/sshubuntu/api/SshUbuntuContainer.kt
Comment (pczuj) [inline src/main/kotlin/com/atlassian/performance/tools/sshubuntu/api/SshUbuntuContainer.kt:20]: I'm not sure if this `SshUbuntuProperties` and it's `app.properties` loaded from JAR resources shouldn't be treated as API. It's kinda global configuration mechanism of this class.

Might be worth checking what did we promise in the CHANGELOG.

Comment (pczuj) [reply src/main/kotlin/com/atlassian/performance/tools/sshubuntu/api/SshUbuntuContainer.kt:20]: On the other side we may not be breaking API by not using the values from `app.properties` anymore. I'm not sure.

File: src/main/resources/docker/Dockerfile.template
Comment (mgrzaslewicz) [reply src/main/resources/docker/Dockerfile.template:1]: It's simple enough not to include the dependency. Like including whole http library or just implementing one function

File: src/main/kotlin/com/atlassian/performance/tools/sshubuntu/api/SshUbuntuContainer.kt
Comment (pczuj) [inline src/main/kotlin/com/atlassian/performance/tools/sshubuntu/api/SshUbuntuContainer.kt:70]: Should we add comments with argument names? I may be biased here, but I don't find it useful and even it's not obvious for me how should I read it, because it's not something I'm familiar with.

Comment (pczuj) [reply src/main/kotlin/com/atlassian/performance/tools/sshubuntu/api/SshUbuntuContainer.kt:70]: Maybe a better representation would be something like this
```suggestion
        ImageFromDockerfile(
            "ssh-ubuntu", // dockerImageName
            false // deleteOnExit
        )
```

File: src/main/kotlin/com/atlassian/performance/tools/sshubuntu/api/SshUbuntuContainer.kt
Comment (mgrzaslewicz) [reply src/main/kotlin/com/atlassian/performance/tools/sshubuntu/api/SshUbuntuContainer.kt:20]: I haven't found any other usage of app.properties other than extracting the version.
Its purpose was to pull the same image from docker repository as library version.
And now the version is hardcoded so no need to get it at runtime

File: src/main/kotlin/com/atlassian/performance/tools/sshubuntu/api/SshUbuntuContainer.kt
Comment (mgrzaslewicz) [reply src/main/kotlin/com/atlassian/performance/tools/sshubuntu/api/SshUbuntuContainer.kt:70]: I usually use named parameters in kotlin and run intellij intention for that.
For java code you can only use 'Add names in comments to call arguments' and I often find the code easier to read with such names included.
I can remove it if you don't find it useful

Comment (pczuj) [reply src/main/kotlin/com/atlassian/performance/tools/sshubuntu/api/SshUbuntuContainer.kt:70]: I'm not very opinionated here, but I think it would be good to decide on a single approach/algorythom when it comes to adding names to Java class arguments or not doing that. I'm not sure if we can do that as part of this PR.

I think for the purpose of this PR you should decide yourself if it's worth adding or not.

Comment (mgrzaslewicz) [reply src/main/kotlin/com/atlassian/performance/tools/sshubuntu/api/SshUbuntuContainer.kt:70]: If we add it, I suppose it should look similar to kotlin named parameters.
It makes no harm to leave it as it is and later we can discuss the general rule

**Review by pczuj — APPROVED** [review]
I don't see any problems with those changes. 

@dagguh may have more context of the whole library and be able to spot some problems or suggest better solutions, however he may not be able to review today. I'm not sure if you can wait for his review, yet it could be worth. You should decide.

**Comment by dagguh** [comment]
Interestingly, [my infra integration tests do not need `ssh-ubuntu`](https://github.com/atlassian/infrastructure/pull/103/files#diff-7a8a4e8160462c3f4ff8867407021499034f007fe2f89d7f5a6ff096d863b718R96).

**Comment by mgrzaslewicz** [comment]
@dagguh I'm not aware of ssh-ubuntu being used anywhere apart from unit tests in ssh library. If so, it seems unnecessary to keep it as a separate library

**Comment by dagguh** [comment]
It's also [used in infra tests](https://github.com/atlassian/infrastructure/blob/release-4.19.3/build.gradle.kts#L69). I agree, we should remove/archive this lib. We should start with [raw DockerJava code](https://github.com/dagguh/infrastructure/blob/b6a4576d5059c323ad71ba55f9cb5a6151e097e3/src/test/kotlin/com/atlassian/performance/tools/infrastructure/api/DockerInfrastructure.kt#L16-L21) and [consider extracting an abstraction/lib only when we see 3+ repetitions](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)).

**Pull request merged**
